src/Microsoft.Health.Fhir.Core/Features/Search/SearchOptionsFactory.cs

```diff
@@ -43,6 +43,16 @@ public SearchOptions Create(IReadOnlyList<Tuple<string, string>> queryParameters
         }
 
         public SearchOptions Create(string resourceType, IReadOnlyList<Tuple<string, string>> queryParameters)
+        {
+            return Create(null, null, resourceType, queryParameters);
+        }
+
+        public SearchOptions Create(string compartmentType, string compartmentId, IReadOnlyList<Tuple<string, string>> queryParameters)
+        {
+            return Create(compartmentType, compartmentId, null, queryParameters);
+        }
+
+        public SearchOptions Create(string compartmentType, string compartmentId, string resourceType, IReadOnlyList<Tuple<string, string>> queryParameters)
```

Comment:
Do we need this? It's not called by anything and it can be called with the below method with resourceType set to null.